scipy/stats/morestats.py

```diff
@@ -2150,7 +2192,27 @@ def ansari(x, y):
     With a p-value of 0.00628, the test provides strong evidence that
     the scales of the distributions from which the samples were drawn
     are not equal.
+
+    We can use the ``alternative`` parameter to perform one-tailed test.
+    In the above example, the scale of `x1` is greater than `x3` and so
+    the ratio of scales of `x1` and `x3` is greater than 1. This means
+    that the p-value when ``alternative='greater'`` should be near 0 and
+    hence we should be able to reject the null hypothesis:
+
+    >>> ansari(x1, x3, alternative='greater')
+    AnsariResult(statistic=452.0, pvalue=0.0031401394)
+
+    As we can see, the p-value is indeed less than the level of significance.
+    Use of ``alternative='less'`` should thus yield a large p-value
+
+    >>> ansari(x1, x3, alternative='less')
+    AnsariResult(statistic=452.0, pvalue=0.9971491082105786)
     """
+    if alternative not in ('two-sided', 'greater', 'less'):
```

Comment:
I'll go ahead and resolve this immediately, as it doesn't need to be changed, but:
```suggestion
    if alternative not in {'two-sided', 'greater', 'less'}:
```
Doesn't matter in practice, but this is what sets are for. I don't want `set` to feel left out : (

Sometimes I'll do something like:
```python3
alternatives = {'two-sided', 'greater', 'less'}
if alternative not in alternatives:
    raise ValueError(f"'alternative' must be one of {alternatives}.")
```
but that doesn't matter much either with such a small set of options.

Reply:
> Doesn't matter in practice, but this is what sets are for. I don't want `set` to feel left out : (

Sure, why not! `O(log n)` over `O(n)` anytime, lol :rofl:!